scripts/setup/nrfconnect/get_zap.py

```diff
@@ -0,0 +1,131 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# The script parses integrations/docker/images/chip-build/Dockerfile file
+# from the Matter repository and looks for ENV ZAP_VERSION= string to find
+# currently recommended ZAP version. After that the package matching this version
+# and currently used OS is downloaded from the ZAP release packages and extracted
+# in the location given by the user.
+
+import os
+import re
+import wget
+import subprocess
+import platform
+import argparse
+from zipfile import ZipFile
+
+def get_zap_recommended_version():
+    matter_root = os.path.abspath(os.path.join(os.path.dirname(__file__), os.path.normpath('../../..')))
+    zap_version_file = os.path.join(matter_root, 'integrations/docker/images/chip-build/Dockerfile')
+
+    try:
+        with open(os.path.join(zap_version_file), 'r') as f:
+            file_content = f.read()
+            result = re.findall(r'ENV ZAP_VERSION=(.*)\s', file_content)
+
+            if (len(result) == 0):
+                raise RuntimeError("Couldn't find pattern matching ZAP_VERSION.")
+            elif (len(result) > 1):
+                raise RuntimeError("Found multiple patterns matching ZAP_VERSION.")
+            
+            return result[0]
+    except:
+        raise RuntimeError(
+            "Encountered problem when trying to read " + zap_version_file + " file.")
+
+def get_zap_current_version():
+    try:
+        cmd_out = subprocess.check_output(['zap', '--version'])
+        cmd_decoded_out = cmd_out.decode('ascii').strip()
+        version = re.findall(r'Version: (.*)\s', cmd_decoded_out)
+        
+        if (len(version) == 0):
+            raise RuntimeError("Couldn't find pattern matching Version:.")
+        elif (len(version) > 1):
+            raise RuntimeError("Found multiple patterns matching Version:.")
+            
+        return version[0]
+    except:
+        return None
+
+def download_recommended_zap_package(version, package_name, location):    
+    try:
+        print("Trying to download ZAP tool package matching your system and recommended version.")
+        response = wget.download("https://github.com/project-chip/zap/releases/download/" + version + "/" + package_name, out=location)
+        print("\n")
+    except:
+        raise RuntimeError("Invalid URL to download ZAP tool package")
+
+def unzip_zap_package(location, filename):
+    try:
+        zip = ZipFile(location + filename + '.zip')
+        zip.extractall(location + filename)
+        zip.close()
```

Comment:
What about deleting an unnecessary zip file after extraction? Do we still need it?

Reply:
Done.